include/zephyr/drivers/i2c.h

```diff
@@ -396,6 +396,20 @@ struct i2c_target_config {
 	const struct i2c_target_callbacks *callbacks;
 };
 
+/**
+ * @brief Validate that I2C bus is ready.
+ *
+ * @param spec I2C specification from devicetree
+ *
+ * @retval true if the I2C bus is ready for use.
+ * @retval false if the I2C bus is not ready for use.
+ */
+static inline bool i2c_is_ready(const struct i2c_dt_spec *spec)
```

Comment:
i2c_is_ready_dt() for the same reason explained in https://github.com/zephyrproject-rtos/zephyr/pull/52685#discussion_r1036293976

Reply:
Done.